pkg/settings/setting.go

```diff
@@ -448,3 +449,20 @@ func IterateWhitelistedEnvVars(handler func(name, value string)) {
 		}
 	}
 }
+
+// GetMachineProvisionImagePullPolicy will return the pull policy to be used on MachineProvisioning job.
+//
+//	If an invalid value is set it will return the default value v1.PullAlways
+func GetMachineProvisionImagePullPolicy() v1.PullPolicy {
+	switch v1.PullPolicy(MachineProvisionImagePullPolicy.Get()) {
+	case v1.PullAlways:
+		return v1.PullAlways
+	case v1.PullIfNotPresent:
+		return v1.PullIfNotPresent
+	case v1.PullNever:
```

Comment:
While allowing `v1.PullNever` feels like a foot gun, I think it's a good idea to allow the user to set any of the available pull options as they see fit. Plus, in air-gapped scenarios where a registry has not been configured, `PullIfNotPresent` and `Never` will be functionally the same.